mmseg/models/losses/cross_entropy_loss.py

```diff
@@ -83,12 +116,22 @@ def binary_cross_entropy(pred,
                 pred.dim() == 4 and label.dim() == 3), \
             'Only pred shape [N, C], label shape [N] or pred shape [N, C, ' \
             'H, W], label shape [N, H, W] are supported'
+        # `weight` returned from `_expand_onehot_labels`
+        # has been treated for valid (non-ignore) pixels
         label, weight = _expand_onehot_labels(label, weight, pred.shape,
                                               ignore_index)
+    else:
+        # should mask out the ignored elements
+        valid_mask = ((label >= 0) & (label != ignore_index)).float()
+        if weight is not None:
+            weight *= valid_mask
+        else:
+            weight = valid_mask
+    # average loss over non-ignored elements
+    if avg_factor is None:
```

Comment:
```suggestion
    if reduction == 'mean' and avg_factor is None:
```

No need to calculate avg_factor when reduction='sum' and 'none'

Reply:
and avg_non_ignore